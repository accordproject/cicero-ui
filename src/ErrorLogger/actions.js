export const gtZero = input => input > 0;

export const typeSwitchCase = (input) => {
  switch (true) {
    case ('modelError' in input):
      return 'Model';
    case ('parseError' in input):
      return 'Grammar';
    default:
      return 'Unknown';
  }
};

export const keySwitchCase = (input) => {
  switch (true) {
    case ('modelError' in input):
      return input.clauseTemplateId;
    case ('parseError' in input):
      return input.clauseId;
    default:
      return null;
  }
};

export const overalltypeSwitchCase = (input) => {
  switch (true) {
    case ('modelError' in input):
      return input.modelError;
    case ('parseError' in input):
      return input.parseError;
    default:
      return null;
  }
};

export const errorsExist = errors => gtZero(errors.length);

export const errorArrayLength = errors => (errorsExist(errors) ? errors.length : 'No');

export const isMultpleErrors = errors => ((errors.length > 1) ? 'Errors' : 'Error');

export const truncateMessage = string => ((string.length > 200)
  ? `${string.substring(0, 200)}...` : string);
