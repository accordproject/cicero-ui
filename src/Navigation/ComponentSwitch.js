/* React */
import React from 'react';
import PropTypes from 'prop-types';

/* Styling */
import * as SC from './styles';

/* Constants */
import { NAVIGATION, FILES } from './constants';

/**
 * Represents the navigation between both navigation
 * components, and provides the means to switch between
 * the two components with a state from props
 * @param {*} props
 */
const NavigationComponent = (props) => {
  const navigationProps = {
    navState: props.navState,
    onClick: () => props.setNavState(NAVIGATION),
  };

  const fileProps = {
    navState: props.navState,
    onClick: () => props.setNavState(FILES),
  };

  return (
    <React.Fragment>
        <SC.Navigation {...navigationProps}>
            NAVIGATION
        </SC.Navigation>
        <SC.Files {...fileProps}>
            FILES
        </SC.Files>
    </React.Fragment>
  );
};

NavigationComponent.propTypes = {
  setNavState: PropTypes.func.isRequired,
  navState: PropTypes.string.isRequired
};

export default NavigationComponent;
