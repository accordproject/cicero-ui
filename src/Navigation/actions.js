const isClause = input => input === 'clause';
const isHeadingOne = input => input === 'heading_one';
const isHeadingTwo = input => input === 'heading_two';
const isHeadingThree = input => input === 'heading_three';

const truncateHeading = (heading, length) => ((heading.length > length)
  ? `${heading.substring(0, length)}...` : heading);


export const truncateHeader = ({ type, text }) => {
  if (isClause(type)) return truncateHeading(text, 22);
  if (isHeadingOne(type)) return truncateHeading(text, 22);
  if (isHeadingTwo(type)) return truncateHeading(text, 18);
  if (isHeadingThree(type)) return truncateHeading(text, 14);
  return 'Error!';
};
