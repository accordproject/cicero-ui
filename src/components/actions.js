const titleStart = input => input.lastIndexOf('/');
const titleEnd = input => input.indexOf('@');
const titleReducer = input => input.slice((titleStart(input) + 1), titleEnd(input));
const titleSpacer = input => input.replace(/-/g, ' ');
const titleCaps = input => input.toUpperCase();

export const titleGenerator = (input) => {
  const reducedTitle = titleReducer(input);
  const spacedTitle = titleSpacer(reducedTitle);
  const finalTitle = titleCaps(spacedTitle);
  return finalTitle;
};

export const headerGenerator = (templateTitle, inputTitle) => {
  const title = titleGenerator(templateTitle);
  const header = inputTitle ? (title + inputTitle) : title;
  const truncatedTitle = ((header.length > 54) ? (`${header.slice(0, 54)}...`) : header);
  return truncatedTitle;
};
