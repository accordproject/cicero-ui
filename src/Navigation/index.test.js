import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import NavigationComponent from './index';
import ComponentSwitch from './ComponentSwitch';
import { truncateHeader } from './actions';

describe('<NavigationComponent />', () => {
  const propsFilesTrue = {
    navigationProps: {
      NAVIGATE_SWITCH_FILES_VISIBLE: true
    },
    navigateHeader: jest.fn(),
    headers: []
  };

  const propsFilesFalse = {
    navigationProps: {
      NAVIGATE_SWITCH_FILES_VISIBLE: false
    },
    navigateHeader: jest.fn(),
    headers: []
  };

  describe('on initialization', () => {
    it('renders component correctly showing Files', () => {
      const component = shallow(<NavigationComponent {...propsFilesTrue} />);
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });

    it('renders component correctly hiding Files', () => {
      const component = shallow(<NavigationComponent {...propsFilesFalse} />);
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });

    it('renders without files initially', () => {
      const component = shallow(<NavigationComponent {...propsFilesFalse} />);
      expect(component.find('#NavigationWrapperComponent').exists()).toBeTruthy();
      expect(component.find('#ContractNavigationComponent').exists()).toBeTruthy();
      expect(component.find('#FilesNavigationComponent').exists()).toBe(false);
    });
  });
});

describe('<ComponentSwitch />', () => {
  const propsFilesTrue = {
    filesVisible: true,
    navState: 'FILES',
    setNavState: jest.fn(),
  };

  const propsFilesFalse = {
    filesVisible: false,
    navState: 'NAVIGATION',
    setNavState: jest.fn(),
  };

  describe('on initialization', () => {
    it('renders component correctly showing Files', () => {
      const component = shallow(<ComponentSwitch {...propsFilesTrue} />);
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });

    it('renders component correctly showing Navigation', () => {
      const component = shallow(<ComponentSwitch {...propsFilesFalse} />);
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('truncateHeader', () => {
  it('should truncate a clause heading to 20 characters', () => {
    const clauseHeader = {
      type: 'clause',
      text: 'Acceptance of Delivery',
    };
    const resultClauseHeader = 'Acceptance of Delive...';
    expect(truncateHeader(clauseHeader)).toEqual(resultClauseHeader);
  });

  it('should truncate a heading one to 22 characters', () => {
    const headingOne = {
      type: 'heading_one',
      text: 'Welcome to Template Studio! Edit this text to get started.',
    };
    const resultHeadingOne = 'Welcome to Template St...';
    expect(truncateHeader(headingOne)).toEqual(resultHeadingOne);
  });

  it('should truncate a heading two to 18 characters', () => {
    const headingTwo = {
      type: 'heading_two',
      text: 'Welcome to Template Studio! Edit this text to get started.',
    };
    const resultHeadingTwo = 'Welcome to Templat...';
    expect(truncateHeader(headingTwo)).toEqual(resultHeadingTwo);
  });

  it('should truncate a heading three to 14 characters', () => {
    const headingThree = {
      type: 'heading_three',
      text: 'Welcome to Template Studio! Edit this text to get started.',
    };
    const resultHeadingThree = 'Welcome to Tem...';
    expect(truncateHeader(headingThree)).toEqual(resultHeadingThree);
  });
});
