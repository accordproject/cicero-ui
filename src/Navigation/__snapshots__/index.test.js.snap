// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComponentSwitch /> on initialization renders component correctly showing Files 1`] = `
<Fragment>
  <Styled(styled.a)
    filesVisible={true}
    id="ContractNavigationSwitchComponent"
    navState="FILES"
    onClick={[Function]}
  >
    NAVIGATION
  </Styled(styled.a)>
  <Styled(styled.a)
    filesVisible={true}
    id="ContractFilesSwitchComponent"
    navState="FILES"
    onClick={[Function]}
  >
    FILES
  </Styled(styled.a)>
</Fragment>
`;

exports[`<ComponentSwitch /> on initialization renders component correctly showing Navigation 1`] = `
<Fragment>
  <Styled(styled.a)
    filesVisible={false}
    id="ContractNavigationSwitchComponent"
    navState="NAVIGATION"
    onClick={[Function]}
  >
    NAVIGATION
  </Styled(styled.a)>
  <Styled(styled.a)
    filesVisible={false}
    id="ContractFilesSwitchComponent"
    navState="NAVIGATION"
    onClick={[Function]}
  >
    FILES
  </Styled(styled.a)>
</Fragment>
`;

exports[`<NavigationComponent /> on initialization renders component correctly hiding Files 1`] = `
<styled.div
  id="NavigationWrapperComponent"
>
  <NavigationComponent
    filesVisible={false}
    navState="NAVIGATION"
    setNavState={[Function]}
  />
  <ContractNavigation
    headers={Array []}
    id="ContractNavigationComponent"
    navigateHeader={[MockFunction]}
    navigationProps={
      Object {
        "NAVIGATE_SWITCH_FILES_VISIBLE": false,
      }
    }
    styleProps={
      Object {
        "clauseColor": undefined,
        "clauseHeaderColor": undefined,
        "headerColor": undefined,
      }
    }
  />
</styled.div>
`;

exports[`<NavigationComponent /> on initialization renders component correctly showing Files 1`] = `
<styled.div
  id="NavigationWrapperComponent"
>
  <NavigationComponent
    filesVisible={true}
    navState="NAVIGATION"
    setNavState={[Function]}
  />
  <ContractNavigation
    headers={Array []}
    id="ContractNavigationComponent"
    navigateHeader={[MockFunction]}
    navigationProps={
      Object {
        "NAVIGATE_SWITCH_FILES_VISIBLE": true,
      }
    }
    styleProps={
      Object {
        "clauseColor": undefined,
        "clauseHeaderColor": undefined,
        "headerColor": undefined,
      }
    }
  />
</styled.div>
`;
